node/full_client_test.go

```diff
@@ -380,6 +384,69 @@ func TestGetCommit(t *testing.T) {
 	})
 }
 
+func TestCometBFTLightClientCompability(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+	mockApp, rpc := getRPC(t)
+	mockApp.On("FinalizeBlock", mock.Anything, mock.Anything).Return(finalizeBlockResponse)
+	mockApp.On("Commit", mock.Anything, mock.Anything).Return(&abci.ResponseCommit{}, nil)
+
+	// creat 3 consecutive signed blocks
+	block1, privKey := types.GetRandomBlockWithKey(1, 1, nil)
+
+	block2 := types.GetRandomNextBlock(block1, privKey, []byte{}, 2)
+
+	block3 := types.GetRandomNextBlock(block2, privKey, []byte{}, 3)
+
+	blocks := []*types.Block{block1, block2, block3}
+
+	// start the node for testing
+	err := rpc.node.Start()
+	require.NoError(err)
+	defer func() {
+		assert.NoError(rpc.node.Stop())
+	}()
+	ctx := context.Background()
+
+	// save the 3 blocks
+	for _, b := range blocks {
+		err = rpc.node.Store.SaveBlock(ctx, b, &b.SignedHeader.Commit)
+		rpc.node.Store.SetHeight(ctx, b.Height())
+		require.NoError(err)
+	}
+
+	// Check if the block header provided by rpc.Commit() can be verified using tendermint light client
+	t.Run("checking rollkit ABCI header verifiability", func(t *testing.T) {
+		// trusted header to verify against
+		var trustedHeader cmtypes.SignedHeader
+		setTrustedHeader := false
+		// valset of single sequencer is constant
+		fixedValSet := block1.SignedHeader.Validators
+
+		// for each block (except block 1), verify it's ABCI header with previous block's ABCI header as trusted header
+		for _, b := range blocks {
+			h := int64(b.Height())
+			commit, err := rpc.Commit(context.Background(), &h)
+			require.NoError(err)
+			require.NotNil(commit)
+			require.NotNil(commit.SignedHeader)
+			assert.Equal(h, commit.Height)
+
+			// if there's no trusted header to verify against, set trusted header and skip verifying
+			if !setTrustedHeader {
+				trustedHeader = commit.SignedHeader
+				setTrustedHeader = true
+				continue
+			}
+
+			err = light.Verify(&trustedHeader, fixedValSet, &commit.SignedHeader, fixedValSet, 3*time.Hour, b.Time(), 10*time.Second, cmtmath.Fraction{Numerator: 2, Denominator: 1})
```

Comment:
[nit] can we define variables for 3hrs, 10seconds, and the Fraction so it is easier in the future to understand what we are passing into this method?

Reply:
fixed !